third_party/parser/cc/grammars/typedruby.ypp

```diff
@@ -1955,6 +1961,7 @@ opt_block_args_tail:
                     }
                   f_larglist
                     {
+                      driver.lex.context.pop();
                       driver.lex.cmdarg.push(false);
                     }
                   lambda_body
```

Comment:
https://github.com/whitequark/parser/pull/709/files#diff-7a330c23c558160f33721dcac766f436L2096-R2105

^ why doesn't the Sorbet change have something that looks like these lines?

Reply:
actually i'm having a hard time matching this PR up with the original whitequark PR. it's probably fine.

Reply:
Wouldn't that be that change: https://github.com/sorbet/sorbet/pull/3363/files#diff-7be0b959d201832b58982ab17587c439L1487-R1491